src/caches/extrapolation_caches.jl

```diff
@@ -47,3 +47,258 @@ function alg_cache(alg::RichardsonEuler,u,rate_prototype,uEltypeNoUnits,uBottomE
+  n_curr::Int64 # Storage for the current extrapolation order
+  n_old::Int64 # Storage for the extrapolation order n_curr before perfom_step! changes the latter
+
+  # Constant values
+  subdividing_sequence::Array{BigInt,1}  # subdividing_sequence[n] is used for the (n -1)th internal discretisation
+  stage_number::Array{Int,1} # Stage_number[n] contains information for extrapolation order (n + alg.n_min - 1)
+  # Weights and Scaling factors for extrapolation operators
+  extrapolation_weights::Array{Rational{BigInt},2}
+  extrapolation_scalars::Array{Rational{BigInt},1}
+  # Weights and scaling factors for internal extrapolation operators (used for error estimate)
+  extrapolation_weights_2::Array{Rational{BigInt},2}
+  extrapolation_scalars_2::Array{Rational{BigInt},1}
+end
+
+function alg_cache(alg::ExtrapolationMidpointDeuflhard,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{true}})
+  u_tilde = zero(u)
+  u_temp1 = zero(u)
+  u_temp2 = zero(u)
+  tmp = zero(u)
+  T = fill(zero(u), alg.n_max + 1)
+  res = uEltypeNoUnits.(zero(u))
+  fsalfirst = zero(rate_prototype)
+  k = zero(rate_prototype)
+
+  cc = alg_cache(alg,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,Val{false})
+
+  ExtrapolationMidpointDeuflhardCache(u_tilde, u_temp1, u_temp2, tmp, T, res, fsalfirst, k,
+      cc.Q, cc.n_curr, cc.n_old, cc.subdividing_sequence, cc.stage_number, cc.extrapolation_weights, cc.extrapolation_scalars, cc.extrapolation_weights_2, cc.extrapolation_scalars_2)
+end
+
+@cache mutable struct ExtrapolationMidpointHairerWannerConstantCache{QType} <: OrdinaryDiffEqConstantCache
+  # Values that are mutated
+  Q::Vector{QType} # Storage for stepsize scaling factors. Q[n] contains information for extrapolation order (n - 1)
+  n_curr::Int64 # Storage for the current extrapolation order
+  n_old::Int64 # Storage for the extrapolation order n_curr before perfom_step! changes the latter
+
+  sigma::Rational{Int64} # Parameter for order selection
+
+  # Constant values
+  subdividing_sequence::Array{BigInt,1} # subdividing_sequence[n] is used for the (n -1)th internal discretisation
+  stage_number::Array{Int,1} # Stage_number[n] contains information for extrapolation order (n - 1)
+  # Weights and Scaling factors for extrapolation operators
+  extrapolation_weights::Array{Rational{BigInt},2}
+  extrapolation_scalars::Array{Rational{BigInt},1}
+  # Weights and scaling factors for internal extrapolation operators (used for error estimate)
+  extrapolation_weights_2::Array{Rational{BigInt},2}
+  extrapolation_scalars_2::Array{Rational{BigInt},1}
+end
+
+function alg_cache(alg::ExtrapolationMidpointHairerWanner,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
```

Comment:
I feel like some refactoring can make this re-use a lot of code from the previous cache function. That's a second order issue that we can handle later though.

Reply:
That's true. Am I "allowed" define a function producing the weights etc. somewhere? If so where?

Reply:
just do it in another part of this file, and then call it from in here.

Reply:
Ok I will change that!